### PR TITLE
[Feature][Style] Configure Spotless to format SQL code in DS 

### PR DESCRIPTION
**Pull request by EricGao888**
## Purpose of the pull request

* Configure Spotless to format SQL code in DS. 

## Brief change log

* Configure Spotless to format SQL code in DS.
* Refactor `dolphinscheduler-dao/src/main/resources/sql/dolphinscheduler_mysql.sql` as an example.
* Since the number of SQL scripts in DS is not that big, if it looks good, I will fix the format of legacy sql scripts.

## Verify this pull request

* Verified by manual tests.

### CODE REVIEW

**Comment by EricGao888** [comment]
> ⚠️ This PR do not change database DDL synchronize.

Not sure what this means. I just changed the format, not content : )

**Comment by EricGao888** [comment]
@kezhenxu94 @caishunfeng @SbloodyS Could u plz take a look when available? Thanks~

File: dolphinscheduler-dao/src/main/resources/sql/dolphinscheduler_mysql.sql [hunk 1]
```diff
@@ -14,1955 +14,6933 @@
...
+    TABLE
+        ` QRTZ_CALENDARS `(
+            ` SCHED_NAME ` VARCHAR(120) NOT NULL,
+            ` CALENDAR_NAME ` VARCHAR(200) NOT NULL,
+            ` CALENDAR ` BLOB NOT NULL,
+            PRIMARY KEY(
+                ` SCHED_NAME `,
+                ` CALENDAR_NAME `
+            )
+        ) ENGINE = InnoDB DEFAULT CHARSET = utf8;
```
Comment (SbloodyS) [inline dolphinscheduler-dao/src/main/resources/sql/dolphinscheduler_mysql.sql]: This format is really ugly...

Comment (EricGao888) [reply dolphinscheduler-dao/src/main/resources/sql/dolphinscheduler_mysql.sql]: LOL. This is formatted by [dbeaver](https://github.com/diffplug/spotless/tree/main/plugin-maven#dbeaver), we may choose other formatting options, or switch to use [prettier](https://github.com/diffplug/spotless/tree/main/plugin-maven#prettier)... 🤣

Comment (EricGao888) [reply dolphinscheduler-dao/src/main/resources/sql/dolphinscheduler_mysql.sql]: Although it looks a bit strange, it helps the SQL code more readable once you get used to it. 🤣

Comment (SbloodyS) [reply dolphinscheduler-dao/src/main/resources/sql/dolphinscheduler_mysql.sql]: I still think this is not as good as before.

Comment (EricGao888) [reply dolphinscheduler-dao/src/main/resources/sql/dolphinscheduler_mysql.sql]: > I still think this is not as good as before.

LOL I'm looking into `prettier` to see if we could have a better one. But at least it unifies the sql formatting...

Comment (EricGao888) [reply dolphinscheduler-dao/src/main/resources/sql/dolphinscheduler_mysql.sql]: @SbloodyS @kezhenxu94 @caishunfeng  I've tested [prettier](https://github.com/diffplug/spotless/tree/main/plugin-maven#prettier) with `Spotless`, it seems there's some kind of bug to use `prettier-sql-plugin` with `Spotless`. Thus, `dbeaver` formatter is our only choice with `Spotless` for `SQL` formatting at this moment. We have to trade-off here. If consistency of formatting is more important, we will pick this `dbeaver formatter`. If you think the style is too ugly and prefer not to use it, we just leave those `SQL` scripts unformatted at present. WDYT?

File: style/dbeaver.properties [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+# limitations under the License.
+#
+# case of the keywords (UPPER, LOWER or ORIGINAL)
+sql.formatter.keyword.case=UPPER
+# Statement delimiter
+sql.formatter.statement.delimiter=;
+# Indentation style (space or tab)
+sql.formatter.indent.type=space
+# Number of identation characters
+sql.formatter.indent.size=4
```
Comment (kezhenxu94) [inline style/dbeaver.properties]: Maybe change the indent size to 2?

Comment (EricGao888) [reply style/dbeaver.properties]: @kezhenxu94 Done, it looks better to me.

**Comment by EricGao888** [comment]
If there are no more follow-ups, I will temporarily close this PR since some developers may not get used to this kind of SQL formatting. I will get back to this when `Spotless` support better SQL formatting functions, thanks.

**Comment by zhongjiajie** [comment]
> If there are no more follow-ups, I will temporarily close this PR since some developers may not get used to this kind of SQL formatting. I will get back to this when `Spotless` support better SQL formatting functions, thanks.

Yeah, I think this format is a little odd for me, maybe we should keep it in old, or find other formatting tools

**Comment by EricGao888** [comment]
This tool is under development. Once it is completed, we might use it with `Spotless`. https://www.npmjs.com/package/prettier-plugin-sql

**Pull request closed without merging**